libraries/rush-lib/src/utilities/InteractiveUpgradeUI.ts

```diff
@@ -0,0 +1,187 @@
+import colors from 'colors/safe';
+import inquirer from 'inquirer';
+import type * as inquirerTypes from 'inquirer';
+import table from 'text-table';
+import _ from 'lodash';
+import Separator from 'inquirer/lib/objects/separator';
+
+export interface IUIGroup {
+  title: string;
+  bgColor?: string;
+  filter: {
+    mismatch?: boolean;
+    bump?: undefined | 'major' | 'minor' | 'patch' | 'nonSemver';
+    notInstalled?: boolean;
+  };
+}
+
+export interface IUpgradeInteractiveDepChoice {
+  value: NpmCheck.INpmCheckPackage;
+  name: string | string[];
+  short: string;
+}
+
+type ChoiceTable = (Separator | IUpgradeInteractiveDepChoice | boolean | undefined)[] | undefined;
+
+// Taken from ansi-regex node module
+// todo: find out how to use the ansi-regex esmodule without node failures
+const ansiRegex = ({ onlyFirst = false } = {}): RegExp => {
```

Comment:
Just because ECMAScript now has lambdas doesn't mean good old `function` doesn't still have its place for things like this. The scope hoisting magic of `function` also helps avoid definition ordering issues, and when dealing with `typedef` lint rules, avoids repeating yourself.